lib/dataset/dataset_operations_common.h

```diff
@@ -78,7 +78,7 @@ template <class Func, class... Args>
 Dataset apply_to_items(const Dataset &d, Func func, Args &&...args) {
   Dataset result;
   for (const auto &data : d)
-    result.setData(data.name(), func(data, std::forward<Args>(args)...));
+    result.setDataInit(data.name(), func(data, std::forward<Args>(args)...));
   return result;
 }
```

Comment:
Will return invalid dataset if input is valid but has no items? See comment above... should we init with dims instead?